src/app/app.component.ts

```diff
@@ -31,71 +34,54 @@ export class AppComponent implements AfterViewInit, OnInit, OnDestroy {
   ];
 
   @ViewChild(AlertComponent, {static: false})
-  alert: AlertComponent | undefined;
-
-  alertSubject$: BehaviorSubject<boolean> = new BehaviorSubject(false);
```

Comment:
For the time being, I removed Subject and Subscription code. If, for some reason, one would like to keep track of the opened/closed state of an overlay, using `signals` would be much better here, anyway.